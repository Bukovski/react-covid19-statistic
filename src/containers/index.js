export { default as SettingsContainer } from './settings.container';
export { default as InfoContainer } from './info.container';
