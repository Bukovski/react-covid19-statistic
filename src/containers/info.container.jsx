import React, { useState, useEffect, Fragment } from 'react';
import { fetchStatisticData } from "../api";
import { Info, Chart, Countries, Spinner } from '../components';


const InfoContainer = () => {
	const [ data, setData ] = useState({});
	const [ country, setCountry ] = useState('');
	
	useEffect(() => {
		const fetchAPI = async () => {
			const dataFetch = await fetchStatisticData();

			setData(dataFetch);
		};
		
		fetchAPI();
	}, []);
	
	const handleCountryChange = async (event) => {
		const country = event.target.value;
		const data = await fetchStatisticData(country);
		
		setData(data);
		setCountry(country);
	};
	
	if (data && !data.confirmed) return <Spinner />;
	
	return (
		<Fragment>
			<Info data={ data }/>
			<Countries handleCountryChange={ handleCountryChange }/>
			<Chart data={ data } country={ country } />
		</Fragment>
	)
};


export default InfoContainer;
