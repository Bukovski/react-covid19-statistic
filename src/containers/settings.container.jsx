import React, { useState, useEffect } from 'react';
import i18next from 'i18next';
import { Settings } from "../components";
import useLocalStorageSetting from "../hooks/useLocalStorageSetting.container";


const SettingsContainer = ({ onToggleUpdate }) => {
	const [ language, setLanguage ] = useState('ru');
	const [ theme, setTheme ] = useState('light');
	const [ localSetting, setLocalSetting ] = useLocalStorageSetting({});
	
	useEffect(() => {
			languageToggle(localSetting.language);
			themeToggle(localSetting.theme);
		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, []);
	
	const languageToggle = (language) => {
		i18next.init({
			lng: language,
			resources: require(`../assets/language/${language}.json`)
		});
		
		setLanguage(i18next.language);
	};
	
	const backgroundSwitcher = () => {
		const getCssBackgroundVar = window
			.getComputedStyle(document.documentElement)
			.getPropertyValue("--primaryBackground");
		
		document.body.style.backgroundColor = getCssBackgroundVar;
	}
	
	const themeToggle = (theme) => {
		document.documentElement.setAttribute('data-theme', theme);
		
		setTheme(theme);
		
		backgroundSwitcher();
	};
	
	
	const handleLanguageChange = (event) => {
		const language = event.target.value;
		
		setLocalSetting({ language: language });
		
		languageToggle(language);
		
		onToggleUpdate();
	};
	
	const handleThemeChange = (event) => {
		const theme = event.target.value;
		
		setLocalSetting({ theme: theme });

		themeToggle(theme);
	};
	

	return (<Settings>
		<Settings.Language
			language={ language }
			handleLanguageChange={ handleLanguageChange }
		/>
		<Settings.Theme
			theme={ theme }
			handleThemeChange={ handleThemeChange }
		/>
	</Settings>)
}


export default SettingsContainer;
