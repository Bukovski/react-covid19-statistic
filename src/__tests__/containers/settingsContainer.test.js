import React from 'react';
import { render, screen } from "@testing-library/react";
import { SettingsContainer } from "../../containers";
import expect from "expect";
import userEvent from "@testing-library/user-event";


describe("SettingsContainer", () => {
	it("change page language", async () => {
		const handleToggleUpdate = jest.fn();
		
		const { container } = render(<SettingsContainer onToggleUpdate={ handleToggleUpdate }/>)
		
		expect(screen.getByText("Изменить язык")).toBeTruthy();
		expect(container.firstChild).toMatchSnapshot();
		
		const languageDrop = container.getElementsByTagName("select")[ 0 ];
		await userEvent.selectOptions(languageDrop, "en");
		
		expect(screen.getByText("Change Language")).toBeTruthy();

		expect(handleToggleUpdate).toHaveBeenCalled();
		expect(container.firstChild).toMatchSnapshot();
	})
	
	it("change page theme color", async () => {
		const { container } = render(<SettingsContainer/>)
		
		expect(screen.getByText("Light").selected).toBeTruthy();
		expect(screen.getByText("Dark").selected).toBeFalsy();
		
		const themeDrop = container.getElementsByTagName("select")[ 1 ];
		await userEvent.selectOptions(themeDrop, "dark");
		
		expect(screen.getByText("Light").selected).toBeFalsy();
		expect(screen.getByText("Dark").selected).toBeTruthy();
	})
})
