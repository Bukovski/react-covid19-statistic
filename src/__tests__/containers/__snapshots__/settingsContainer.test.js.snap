// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsContainer change page language 1`] = `
<div
  class="settings-container"
>
  <div
    class="MuiFormControl-root formControl settings-language"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      data-testid="test-language"
    >
      <select
        aria-invalid="false"
        class="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input"
      >
        <option
          value="ru"
        >
          RU
        </option>
        <option
          value="en"
        >
          EN
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiNativeSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
    <p
      class="MuiFormHelperText-root settings-text MuiFormHelperText-filled"
    >
      Изменить язык
    </p>
  </div>
  <div
    class="MuiFormControl-root formControl settings-language"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <select
        aria-invalid="false"
        class="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input"
      >
        <option
          value="light"
        >
          Light
        </option>
        <option
          value="dark"
        >
          Dark
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiNativeSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
    <p
      class="MuiFormHelperText-root settings-text MuiFormHelperText-filled"
    >
      Изменить цвет
    </p>
  </div>
</div>
`;

exports[`SettingsContainer change page language 2`] = `
<div
  class="settings-container"
>
  <div
    class="MuiFormControl-root formControl settings-language"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline Mui-focused Mui-focused MuiInputBase-formControl MuiInput-formControl"
      data-testid="test-language"
    >
      <select
        aria-invalid="false"
        class="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input"
      >
        <option
          value="ru"
        >
          RU
        </option>
        <option
          value="en"
        >
          EN
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiNativeSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
    <p
      class="MuiFormHelperText-root settings-text MuiFormHelperText-filled Mui-focused"
    >
      Change Language
    </p>
  </div>
  <div
    class="MuiFormControl-root formControl settings-language"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <select
        aria-invalid="false"
        class="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input"
      >
        <option
          value="light"
        >
          Light
        </option>
        <option
          value="dark"
        >
          Dark
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiNativeSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
    <p
      class="MuiFormHelperText-root settings-text MuiFormHelperText-filled"
    >
      Change color
    </p>
  </div>
</div>
`;
